borg/cache.py

```diff
@@ -375,12 +378,12 @@ def add_chunk(self, id, data, stats, overwrite=False):
         data = self.key.encrypt(data)
         csize = len(data)
         self.repository.put(id, data, wait=False)
-        self.chunks[id] = (refcount + 1, size, csize)
+        self.chunks[id] = ChunkIndexEntry(refcount + 1, size, csize)
```

Comment:
chunks.add does not set size and csize.
if we force overwrite, we likely have a reason, e.g. changed compression or encryption.
so the same id chunk may have different csize.